jest-assessment-week-2-borja/code-challenges.test.js

```diff
@@ -0,0 +1,315 @@
+// ASSESSMENT 2: Coding Practical Questions with Jest
+
+// Please read all questions thoroughly
+// Pseudo coding is REQUIRED
+// If you get stuck, please leave comments to help us understand your thought process
+
+// Use test driven development to complete the following questions
+// Add appropriate dependencies: $ yarn add jest
+
+// Reminder: The test will call your function
+// Run the file with the following command: $ yarn jest
+
+// --------------------INSTRUCTOR EXAMPLE: Create a function that takes in an array of numbers and returns an array with all the numbers multiplied by 3.
+
+// a) Create a test with expect statements for each of the variables provided.
+
+const numbersArray1 = [6, 7, 8, 9, 10]
+// Expected output: [18, 21, 24, 27, 30]
+const numbersArray2 = [24, 27, 30, 33, 36]
+// Expected output: [72, 81, 90, 99, 108]
+
+// b) Create the function that makes the test pass.
+
+//------------------------------------ Assessment Start ------------------------------------
+
+// jest test template:
+
+// describe ("nameOfFunction", () => {
+//     it("explain what it takes and return expectation in plain English as this is for the programmer", () => {
+//         expect(nameOfFunction()).toEqual(expected return)
+//     })
+// })
+
+// object template:
+
+// const objectName = { key: value }
+
+// --------------------1) Create a function that takes an object as an argument and decides if the number inside it, is evenly divisible by three or not.
+
+// a) Create a test with expect statements for each of the variables provided.
+
+const object1 = { number: 15 }
+// Expected output: "15 is divisible by three"
+const object2 = { number: 0 }
+// Expected output: "0 is divisible by three"
+const object3 = { number: -7 }
+// Expected output: "-7 is not divisible by three"
+
+//------------------------------------ My Test Code ------------------------------------
+
+//pseudo code:
+// I will use the jest test template to test a function "divisibleByThree"
+// it "takes an object { number: 2} and returns if the number is divisible by three"
+// expect(divisibleByThree({number : 15})).toEqual("15 is divisible by three.")
+// expect(divisibleByThree({number : 0})).toEqual("0 is divisible by three.")
+// expect(divisibleByThree({number : -1})).toEqual("-7 is not divisible by three.")
+// input: since there isn't a function to test, there is no input and I expect a fail (red)
+// output: should output a ReferenceError: divisibleByThree is not defined
+ 
+describe ("divisibleByThree", () => {
+    it ("takes an object { number: 2} and returns if the number is divisible by three", () => {
+        expect(divisibleByThree({number : 15})).toEqual("15 is divisible by three.")
```

Comment:
Test is set up nicely, you can use the variables just like you would when invoking a function and passing in the argument.